File: smart-fridge-java-backend/src/main/java/css/ch/smartfridgejavabackend/user/mapper/UserMapper.java
package css.ch.smartfridgejavabackend.user.mapper;

import css.ch.smartfridgejavabackend.auth.dto.SignupRequest;
import css.ch.smartfridgejavabackend.user.dto.UserResponseDTO;
import css.ch.smartfridgejavabackend.user.model.User;
import org.springframework.stereotype.Component;

@Component
public class UserMapper {

    public User map(SignupRequest signupRequest) {
        if (signupRequest == null) {
            throw new IllegalArgumentException("SignupRequest darf nicht null sein");
        }

        User user = new User();
        user.setName(signupRequest.name());
        user.setEmail(signupRequest.email());
        return user;
    }

    public UserResponseDTO map(User user) {
        if (user == null) {
            throw new IllegalArgumentException("User darf nicht null sein");
        }

        return UserResponseDTO.builder()
                .id(user.getId())
                .name(user.getName())
                .email(user.getEmail())
                .build();
    }
}

File: smart-fridge-backend/src/main/kotlin/ch/bbzw/smartfridgebackend/user/mapper/UserMapper.kt
package ch.bbzw.smartfridgebackend.user.mapper

import ch.bbzw.smartfridgebackend.auth.dto.SignupRequest
import ch.bbzw.smartfridgebackend.user.dto.UserResponse
import ch.bbzw.smartfridgebackend.user.model.User
import org.springframework.stereotype.Component

@Component
class UserMapper {
    fun map(signupRequest: SignupRequest): User {
        return User(
            name = signupRequest.name,
            email = signupRequest.email,
            password = ""
        )
    }

    fun map(user: User): UserResponse {
        return UserResponse(
            id = user.id ?: "",
            name = user.name,
            email = user.email
        )
    }
}